### PR TITLE
chore: Automate the prepended licensed header using github actions for the public repository

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes https://jira.tools.sap/browse/CXSPA-1083

### CODE REVIEW

File: .github/workflows/prepend-license.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+on:
+  push:
+    branches:
+      - develop
```
Comment (bgambocjaviniar) [inline .github/workflows/prepend-license.yml]: develop and release branch will trigger this workflow, which will push a new commit if there's any files that needs to be licensed.

Currently, this PR ran the tool beforehand, which is why there's other changes.
I can remove it if you want

File: .github/workflows/prepend-license.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+name: Prepend licences to files
+jobs:
+  prepend_license: 
+    name: Commit prepended licences to files 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GH_ADMIN }}
```
Comment (bgambocjaviniar) [inline .github/workflows/prepend-license.yml]: needs the PAT of an admin to bypass the protected branch.

Comment (RadhepS) [reply .github/workflows/prepend-license.yml]: Could this be dangerous in that anyone can modify the script and run any git command on develop?

File: abc.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+console.log('abc');
```
Comment (RadhepS) [inline abc.ts]: delete file

File: .github/workflows/prepend-license.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+name: Prepend licences to files
+jobs:
+  prepend_license: 
+    name: Commit prepended licences to files 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.GH_PR_TOKEN }}
```
Comment (bgambocjaviniar) [inline .github/workflows/prepend-license.yml:14]: TODO: need to get a token

also, for anyone picking this up, they can test it directly on this branch

Comment (bgambocjaviniar) [reply .github/workflows/prepend-license.yml:14]: done

**Review by RadhepS — APPROVED** [review]

**Pull request merged**